Sources/SwiftProtobuf/duration.pb.swift

```diff
@@ -117,10 +117,16 @@ public struct Google_Protobuf_Duration: SwiftProtobuf.Message, SwiftProtobuf.Pro
 
   public init() {}
 
-  public mutating func _protoc_generated_decodeField<T: SwiftProtobuf.FieldDecoder>(setter: inout T, protoFieldNumber: Int) throws {
-    switch protoFieldNumber {
-    case 1: try setter.decodeSingularField(fieldType: SwiftProtobuf.ProtobufInt64.self, value: &seconds)
-    case 2: try setter.decodeSingularField(fieldType: SwiftProtobuf.ProtobufInt32.self, value: &nanos)
+  public mutating func _protoc_generated_decodeMessage<D: SwiftProtobuf.Decoder>(decoder: inout D) throws {
```

Comment:
I plan to look into moving this boilerplate into a library extension.  It will be interesting to see what performance and/or code size impact that has.